### PR TITLE
Update to CCCL 2.7.0-rc2.

### PR DESCRIPTION
**Pull request by bdice**
This PR updates to CCCL 2.7.0-rc2. Do not merge until all of RAPIDS is ready to update.

Depends on https://github.com/rapidsai/rapids-cmake/pull/710 and should be admin-merged immediately after that PR.

Part of https://github.com/rapidsai/build-planning/issues/115.

### CODE REVIEW

File: cpp/cmake/thirdparty/get_cccl.cmake
Comment (bdice) [inline cpp/cmake/thirdparty/get_cccl.cmake:39]: @robertmaynard I'd appreciate your eyes on this. There was an upstream change in CCCL in https://github.com/NVIDIA/cccl/pull/2310 that made it possible to force CCCL to only compile kernels with 32-bit offsets when `THRUST_DISPATCH_TYPE` is set to `"Force32bit"`. However, I do not know how to forward this argument through `rapids_cpm_cccl`. If I were using `rapids_cpm_find` I think I could use `OPTIONS`. What's the best approach?

Secondly, I am wondering if CCCL should be using `FORCE` here: https://github.com/NVIDIA/cccl/blob/264b587d0621ee2c6bec27aa015078056cb5f021/thrust/CMakeLists.txt#L38
Does that override my ability to set the variable ahead of time, or not since it's marked as `CACHE`?

Comment (robertmaynard) [reply cpp/cmake/thirdparty/get_cccl.cmake:39]: Yes the `FORCE` means that we can't overwrite as each time CMake is invoked the value will be reset to `Force32bit`.

As for passing this down from `rapids_cpm_cccl` you would do `rapids_cpm_cccl(CPM_ARGS OPTIONS "THRUST_DISPATCH_TYPE Force32bit")`

Comment (bdice) [reply cpp/cmake/thirdparty/get_cccl.cmake:39]: https://github.com/NVIDIA/cccl/pull/2844 exposed the dispatch mode in `thrust_create_target` but I don't see how we are calling that in rapids-cmake. For now I am going to try and move forward with `2.7.0-rc2` as-is rather than trying to backport this with a patch. We can adopt that dispatch option in a later CCCL update.

Comment (bdice) [reply cpp/cmake/thirdparty/get_cccl.cmake:39]: Reverted ce7e991 for now.

File: rapids_config.cmake
Comment (bdice) [inline rapids_config.cmake:17]: This PR will merge after https://github.com/rapidsai/rapids-cmake/pull/710. We must revert these changes before merging this PR.
```suggestion
```

**Comment by bdice** [comment]
This is ready for review by cuDF CMake codeowners.

**Review by robertmaynard — APPROVED** [review]

**Pull request merged**
